internal/cli/setup/cluster_config.go

```diff
@@ -0,0 +1,193 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/pointer"
+	atlasClustersPinned "go.mongodb.org/atlas-sdk/v20240530005/admin"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20250312003/admin"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	defaultNodeCount = 3
+	defaultPriority  = 7
+)
+
+func defaultDiskSizeGB(provider, tier string) float64 {
+	return atlas.DefaultDiskSizeGB[strings.ToUpper(provider)][tier]
+}
+
+// newCluster creates a new cluster for the pinned API version.
+func (opts *Opts) newCluster() *atlasClustersPinned.AdvancedClusterDescription {
+	cluster := &atlasClustersPinned.AdvancedClusterDescription{
+		GroupId:                      pointer.Get(opts.ConfigProjectID()),
+		ClusterType:                  pointer.Get(replicaSet),
+		Name:                         &opts.ClusterName,
+		TerminationProtectionEnabled: &opts.EnableTerminationProtection,
+		ReplicationSpecs: &[]atlasClustersPinned.ReplicationSpec{
+			{
+				NumShards: pointer.Get(1),
+				ZoneName:  pointer.Get("Zone 1"),
+				RegionConfigs: &[]atlasClustersPinned.CloudRegionConfig{
+					opts.newAdvancedRegionConfig(),
+				},
+			},
+		},
+	}
+
+	opts.setTags(cluster)
+
+	if diskSizeGB := opts.getDiskSizeOverride(); diskSizeGB != nil {
+		cluster.DiskSizeGB = diskSizeGB
+	}
+	if version := opts.getVersionOverride(); version != nil {
+		cluster.MongoDBMajorVersion = version
+	}
+
+	return cluster
+}
+
+// setTags sets the tags for a cluster of the pinned API version.
+func (opts *Opts) setTags(cluster *atlasClustersPinned.AdvancedClusterDescription) {
+	if len(opts.Tag) > 0 {
+		var tags []atlasClustersPinned.ResourceTag
+		for k, v := range opts.Tag {
+			if k != "" && v != "" {
+				tags = append(tags, atlasClustersPinned.ResourceTag{Key: k, Value: v})
+			}
+		}
+		cluster.Tags = &tags
+	}
+}
+
+// newAdvancedRegionConfig creates a new advanced region config for the pinned API version.
+func (opts *Opts) newAdvancedRegionConfig() atlasClustersPinned.CloudRegionConfig {
+	providerName := opts.providerName()
+
+	regionConfig := atlasClustersPinned.CloudRegionConfig{
+		ProviderName: &providerName,
+		Priority:     pointer.Get(defaultPriority),
+		RegionName:   &opts.Region,
+	}
+
+	regionConfig.ElectableSpecs = &atlasClustersPinned.HardwareSpec{
+		InstanceSize: &opts.Tier,
+	}
+
+	if providerName == tenant {
+		regionConfig.BackingProviderName = &opts.Provider
+	} else {
+		regionConfig.ElectableSpecs.NodeCount = pointer.Get(defaultNodeCount)
+	}
+
+	return regionConfig
+}
+
+// getDiskSizeOverride returns the disk size override, which is only applicable for non-tenant clusters.
+func (opts *Opts) getDiskSizeOverride() *float64 {
+	if opts.providerName() != tenant {
+		diskSizeGB := defaultDiskSizeGB(opts.providerName(), opts.Tier)
```

Comment:
When you pass an incorrect combination to `defaultDiskSizeGB` the value returned is `0`, shouldn't we check that the value `!=0` before returning?

https://go.dev/play/p/mFYKv99R1VO

Reply:
yes! good idea, and if it's zero we should also return nil, thank you 🙏